### PR TITLE
feat: useSendTransaction, useBalance, useEstimateFees, useStark* & useSignTypedData hook

### PR DESCRIPTION
**Pull request by jaipaljadeja**
- removed `useContractWrite`
 - added strict type safety on `useContract`
 - added `useSendTransaction` hook
 - added `useBalance` hook
 - added `useEstimateFees` hook
 - added `useStarkAddress`, `useStarkName` & `useStarkProfile` (throws some errors)
 - renamed `useWaitForTransaction` to `useTransactionReceipt`
 - some tests and type fixes

### CODE REVIEW

File: packages/core/src/hooks/useContract.ts [hunk 1]
```diff
@@ -1,30 +1,72 @@
...
+type _Contract = {
+  [K in keyof Contract]: Contract[K];
+};
+
+export type StarknetTypedContract<TAbi extends Abi> = TypedContract<TAbi> &
+  Omit<_Contract, "populate" | "populateTransaction" | "call"> & {
+    call<TFunctionName extends ExtractAbiFunctionNames<TAbi>>(
+      method: TFunctionName,
+      args?: FunctionArgs<TAbi, TFunctionName>,
+      options?: CallOptions
```
Comment (jaipaljadeja) [inline packages/core/src/hooks/useContract.ts:22]: defined `call` type explicitly as `abi-wan-kanabi` didnt have options parameter as third argument but `starknet.js` did

File: packages/core/src/hooks/index.ts [hunk 2]
```diff
@@ -4,14 +4,14 @@ export * from "./useBlock";
 export * from "./useBlockNumber";
 export * from "./useConnect";
 export * from "./useContractFactory";
```
Comment (fracek) [inline packages/core/src/hooks/index.ts:6]: We need to export `useCotract` from here too.

File: packages/core/src/hooks/useSendTransaction.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  RequestArgs,
+  RequestResult,
+  UseWalletRequestProps,
+  UseWalletRequestResult,
+  useWalletRequest,
+} from "./useWalletRequest";
+
+type UseSendTransactionArgs = {
+  /** List of smart contract calls to execute. */
+  calls: Call[];
```
Comment (fracek) [inline packages/core/src/hooks/useSendTransaction.ts:13]:
```suggestion
  calls?: Call[];
```

Like other APIs, it should be allowed to be `undefined` and in that case it won't send the transaction.

File: packages/core/src/hooks/useContract.ts (hunk 1 above)
Comment (fracek) [reply packages/core/src/hooks/useContract.ts:22]: Where is `call` used? In `useCall` you cast to `Contract` so I don't think it's needed there right?

**Review by fracek — COMMENTED** [review]
Added some minor comments and questions.

File: packages/core/src/hooks/useContract.ts (hunk 1 above)
Comment (jaipaljadeja) [reply packages/core/src/hooks/useContract.ts:22]: > Where is `call` used? In `useCall` you cast to `Contract` so I don't think it's needed there right?

it's not used anywhere else except in `useCall`  which as you mentioned right that its casted as `Contract` so its of no use for us, but i added it assuming if user ever uses `useContract` directly, and uses its `call` so they can have strict type safety  along with the `options` parameter.
let me know your thoughts, we can remove it if you think its not needed.

**Pull request merged**